src/bench/connection.rs

```diff
@@ -149,16 +158,36 @@ impl Connection {
 
             // println!("Id = {}, {:?}", id, incoming);
 
-            if let Event::Incoming(v) = event {
-                match v {
-                    Incoming::PubAck(_pkid) => acks_count += 1,
-                    Incoming::Publish(_publish) => incoming_count += 1,
-                    Incoming::PingResp => {}
-                    incoming => {
-                        error!("Id = {}, Unexpected incoming packet = {:?}", id, incoming);
-                        break;
+            match event {
+                Event::Incoming(v) => {
+                    match v {
+                        Incoming::PubAck(_pkid) => {
+                            acks_count += 1;
+                            // PubACK received for pkid `x`. Server acknowledged it. 
+                            let publish_time = vector.pop_front().unwrap();
+                            let time_elapsed = Instant::now().duration_since(publish_time).as_millis();
+                            hist.record(time_elapsed as u64).unwrap();
+                        },
+                        Incoming::Publish(_publish) => {
+                            incoming_count += 1;
+                            
+                        },
+                        Incoming::PingResp => {}
+                        incoming => {
+                            error!("Id = {}, Unexpected incoming packet = {:?}", id, incoming);
+                            break;
+                        }
                     }
-                }
+                },
+                Event::Outgoing(v) =>{
```

Comment:
This match branch can be simplified to `Event::Outgoing(Outgoing::Publish(pkid))`